dali/sasha/saxref.cpp

```diff
@@ -670,6 +678,61 @@ class CNewXRefManagerBase
 
     }
 
+    void startHeartbeat(const char * op)
+    {
+        CriticalBlock block(heartbeatSect);
+        time(&heartbeatStartTime);
+        lastHeartbeat = heartbeatStartTime;
+        heartbeatInterval = 60; // Start with 1 minute (in seconds)
+        heartbeatCount = 0;
+        uncondlog("%s heartbeat started (interval: 1 minute)", op);
+    }
+
+    void checkHeartbeat(const char * op)
+    {
+        CriticalBlock block(heartbeatSect);
+        time_t now = time(NULL);
+
+        if ((now - lastHeartbeat) >= heartbeatInterval) {
+            unsigned elapsedMinutes = (unsigned)(now - heartbeatStartTime) / 60;
```

Comment:
It is best to minimize the work done in the critical section - because all other threads will be blocked while this code is executing.  A good pattern would be to update lastHeartbeat, increment count etc., but set flags that control any logging.  Then after leaving the critical section check the flags and perform any logging.